ui/perfherder/graphs/TestDataModal.jsx

```diff
@@ -260,6 +268,9 @@ export default class TestDataModal extends React.Component {
       const { seriesData } = updates;
       const tags = this.getTagOptions(seriesData);
       this.setState({ ...updates, tags });
+      if (tag !== this.defaultTag) {
```

Comment:
I see the following lines of code are duplicated inside `processOptions`, which isn't nice:
```javascript
if (tag !== this.defaultTag) {
    await this.updateTaggedTests(tag);
  }
if (filterText) {
    await this.updateFilterText(filterText);
  }
```
We need to find a way to DRY them.